programs/ziti-edge-tunnel/windows-service.c

```diff
@@ -453,7 +450,7 @@ DWORD LphandlerFunctionEx(
 
     switch (dwControl) {
         case SERVICE_CONTROL_STOP:
-            ReportSvcStatus(SERVICE_STOP_PENDING, NO_ERROR, 0);
+            ReportSvcStatus(SERVICE_STOP_PENDING, NO_ERROR, 60000);
 
             // stops the running tunnel service
             scm_service_stop();
```

Comment:
This scm thread just calls the uv_stop function on the loop. When we called this scm_service_stop function using async, the service got stopped, but the scm handler didnt wait for the stop request sent by the scm_service_stop to the handler. So the scm handler threw an error.

Reply:
calling any `uv_` function that is not documented as thread safe must be called from the loop's thread. Doing otherwise creates a race condition on the loop.

Reply:
Will try to block the SCM thread until it receives the stopped event as per our discussion yesterday